test/rewrites.lean

```diff
@@ -51,59 +52,50 @@ info: Try this: rw [← @Nat.prime_iff]
 lemma prime_of_prime (n : ℕ) : Prime n ↔ Nat.Prime n := by
   rw?
 
+#guard_msgs(drop info) in
 example [Group G] (h : G) (hyp : g * 1 = h) : g = h := by
   rw? at hyp
   assumption
 
+#guard_msgs(drop info) in
 example : ∀ (x y : ℕ), x ≤ y := by
   intros x y
   rw? -- Used to be an error here https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/panic.20and.20error.20with.20rw.3F/near/370495531
-  admit
+  exact test_sorry
 
 example : ∀ (x y : ℕ), x ≤ y := by
   -- Used to be a panic here https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/panic.20and.20error.20with.20rw.3F/near/370495531
   success_if_fail_with_msg "Could not find any lemmas which can rewrite the goal" rw?
-  admit
+  exact test_sorry
 
 axiom K : Type
 @[instance] axiom K.ring : Ring K
 
-def foo : K → K := sorry
+noncomputable def foo : K → K := test_sorry
 
+#guard_msgs(drop info) in
 example : foo x = 1 ↔ ∃ k : ℤ, x = k := by
   rw? -- Used to panic, see https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/panic.20and.20error.20with.20rw.3F/near/370598036
-  admit
+  exact test_sorry
 
-lemma six_eq_seven : 6 = 7 := sorry
+lemma six_eq_seven : 6 = 7 := test_sorry
 
 -- This test also verifies that we are removing duplicate results;
 -- it previously also reported `Nat.cast_ofNat`
-/--
-info: Try this: rw [six_eq_seven]
--- ∀ (x : ℕ), x ≤ 7
----
-info: Try this: rw [← @Nat.cast_eq_ofNat]
--- ∀ (x : ℕ), x ≤ ↑6
----
-warning: declaration uses 'sorry'
--/
-#guard_msgs in
+#guard_msgs(drop info) in
 example : ∀ (x : ℕ), x ≤ 6 := by
   rw?
   guard_target = ∀ (x : ℕ), x ≤ 7
-  admit
+  exact test_sorry
 
-example : ∀ (x : ℕ) (w : x ≤ 6), x ≤ 8 := by
+#guard_msgs(drop info) in
+example : ∀ (x : ℕ) (_w : x ≤ 6), x ≤ 8 := by
   rw?
-  guard_target = ∀ (x : ℕ) (w : x ≤ 7), x ≤ 8
-  admit
+  guard_target = ∀ (x : ℕ) (_w : x ≤ 7), x ≤ 8
+  exact test_sorry
 
 -- check we can look inside let expressions
-/--
-info: Try this: rw [@AddCommMonoidWithOne.add_comm]
--- "no goals"
--/
-#guard_msgs in
+#guard_msgs(drop info) in
```

Comment:
@adomani Why are these tests dropping info? Isn't that the point of the test?

Reply:
The info that this is dropping is
```lean
Try this: rw [@AddCommMonoidWithOne.add_comm]
-- "no goals"
```
I was under the impression that this was the desired outcome (see also [this comment](https://github.com/leanprover-community/mathlib4/pull/6868#discussion_r1328670848)).

Reply:
The test is at least checking that the tactic succeeds. Ideally there would be some way to check certain properties of the output, bur for now silencing info nobody was looking at seems fine.

Reply:
But this is the `rewrites.lean` file, it is testing the behavior of `rw?`!

Reply:
Mario, if you want, I can catch the error.  However, does this mean that if someone changes the heuristics that `rw?` uses and `rw?` finds a different lemma to close the goal, then the test will fail?

Reply:
My understanding is that now the test is checking that `rw?` is able to close the goal, regardless of the lemma that it finds to do so.

Reply:
> Mario, if you want, I can catch the error. However, does this mean that if someone changes the heuristics that `rw?` uses and `rw?` finds a different lemma to close the goal, then the test will fail?

Yes, that is desirable, if the heuristics change then the person making the change should be forced to fix the test so that the diff is reviewed

Reply:
So, should I do it for all the `rw?` in this file?

Reply:
I think so, @kmill do you agree?

Reply:
I think testing exact output here isn't great because it depends on the exact lemmas that are imported. We shouldn't be testing that, and people who add lemmas shouldn't have to be responsible for fixing this test.

Reply:
I am happy to go with the flow.  However, my impression is that it would be good to have some test about the output of `rw?`, but testing the *exact* output may not be desirable.

Reply:
We can address that by not importing so much of mathlib and/or creating reasonably representative foobar examples in this test file.

Reply:
I opened #7267, where I re-instate all infos.  Should we move this discussion there?

Reply:
I think we should be sure to distinguish between tests that are checking that the tactic succeeds and tests that are checking the heuristics of the tactic. My impression was that this file was generally testing the first (and small note Damiano: when I'd suggested `drop info`, I only meant it in the places where you added `#guard_msgs` yourself and I had no opinion on the pre-existing ones; I'm fine with the changes that you did make, but it's good to re-review these in #7267 just to be sure what each test is trying to test).

It'd be nice if we had something more sophisticated than `#guard_msgs` to test what we actually want to test... but having tests with a curated set of lemmas to test the heuristics seems very principled.

Reply:
Kyle, sorry that I misunderstood your comment about `drop info`.

I wonder whether the tests about the heuristics of `rw?` should really be testing some of the internals of `rw?`, rather than the `rw?` tactic.